lib/THC/generic/THCTensorRandom.cu

```diff
@@ -265,6 +265,87 @@ THC_API void THCTensor_(multinomial)(struct THCState *state,
   THCTensor_(free)(state, probDistContig);
 }
 
+THC_API void THCTensor_(multinomialAliasSetup)(THCState *state, THCTensor *_probs, THCudaLongTensor *_J, THCTensor *_q){
+  THAssert(THCTensor_(isContiguous)(state, _q));
+  THAssert(THCudaLongTensor_isContiguous(state, _J));
+  THAssert(THCTensor_(isContiguous)(state, _probs));
+  long inputsize = THCTensor_(nElement)(state, _probs);
+  THCudaLongTensor *smaller = THCudaLongTensor_newWithSize1d(state, inputsize);
+  THCudaLongTensor *larger = THCudaLongTensor_newWithSize1d(state, inputsize);
+  THCudaLongTensor *smaller_short = THCudaLongTensor_newWithSize1d(state, inputsize);
+  THCudaLongTensor *larger_short = THCudaLongTensor_newWithSize1d(state, inputsize);
+  
+  THCudaLongTensor_resize1d(state, _J, inputsize);
+  THCudaLongTensor_fill(state, _J, 0);	
```

Comment:
unless i'm wrong, you dont need to fill them with zeros here. You can do that all-together for _J, larger_short, smaller_short in the first CUDA kernel call (over here, is it `aliasMultinomialFilter` i think?)
@pavanky correct me if I'm wrong.